generate.pl

```diff
@@ -91,6 +91,7 @@ sub die_with_sample {
     url    => "https://www.cpan.org/authors/id/M/MI/MIYAGAWA/App-cpanminus-1.7047.tar.gz",
     # sha256 taken from http://www.cpan.org/authors/id/M/MI/MIYAGAWA/CHECKSUMS
     sha256 => "963e63c6e1a8725ff2f624e9086396ae150db51dd0a337c3781d09a994af05a5",
+    patch  => q[perl -pi -E 's{http://(www\.cpan\.org|backpan\.perl\.org|cpan\.metacpan\.org|fastapi\.metacpan\.org|cpanmetadb\.plackperl\.org)}{https://$1}g' bin/cpanm],
```

Comment:
Hi! Afaik cpan.cpantesters.org seems to be in tests and docs only, so not sure it's needed for functionality

Reply:
> Thanks, we are adding a patch in openSUSE as well
 
Just FYI, this patch is safe in this Docker image because it knows ca-certificates and other requirements are there -- the patch doesn't have any fallback, so you likely want to make sure a generated RPM depends on enough components for HTTPS downloads to actually work, so this doesn't break people in minimal containers or other situations.

Reply:
Thanks @perlpunk for the suggestion! 🙏 The cpantesters mentions are only found in pod and help messages, but still worth updating anyway to emphasize HTTPS usage throughout this program.  Will add this in after adding in new tests.